Framework/CurveFitting/src/Functions/CrystalFieldControl.cpp

```diff
@@ -55,10 +55,31 @@ void CrystalFieldControl::setAttribute(const std::string &name,
       m_temperatures = attr.asVector();
       buildControls();
     } else if (name == "FWHMs") {
+      const size_t nSpec = m_temperatures.size();
       m_FWHMs = attr.asVector();
-      if (m_FWHMs.size() == 1 && m_FWHMs.size() != m_temperatures.size()) {
-        m_FWHMs.assign(m_temperatures.size(), m_FWHMs.front());
+      if (m_FWHMs.size() == 1 && m_FWHMs.size() != nSpec) {
+        m_FWHMs.assign(nSpec, m_FWHMs.front());
+      }
+      if (attr.asVector().size() > 0) {
```

Comment:
Could you use `empty()`? `attr.asVector()` creates a new copy of the vector. Can you use `m_FWHMs` instead?